integration_tests/src/main/scala/com/nvidia/spark/rapids/tests/tpch/TpchLikeSpark.scala

```diff
@@ -16,34 +16,58 @@
 
 package com.nvidia.spark.rapids.tests.tpch
 
-import com.nvidia.spark.rapids.tests.DebugRange
+import com.nvidia.spark.rapids.tests.tpch.TpchLikeSpark.{csvToOrc, csvToParquet}
+import org.rogach.scallop.ScallopConf
 
-import org.apache.spark.sql.{DataFrame, SparkSession}
+import org.apache.spark.sql.{DataFrame, DataFrameWriter, Row, SparkSession}
 import org.apache.spark.sql.types._
 
 // scalastyle:off line.size.limit
 
 object TpchLikeSpark {
-  def csvToParquet(spark: SparkSession, basePath: String, baseOutput: String): Unit = {
-    readOrdersCSV(spark, basePath + "/orders.tbl").write.parquet(baseOutput + "/orders.tbl")
-    readLineitemCSV(spark, basePath + "/lineitem.tbl").write.parquet(baseOutput + "/lineitem.tbl")
-    readCustomerCSV(spark, basePath + "/customer.tbl").write.parquet(baseOutput + "/customer.tbl")
-    readNationCSV(spark, basePath + "/nation.tbl").write.parquet(baseOutput + "/nation.tbl")
-    readPartCSV(spark, basePath + "/part.tbl").write.parquet(baseOutput + "/part.tbl")
-    readPartsuppCSV(spark, basePath + "/partsupp.tbl").write.parquet(baseOutput + "/partsupp.tbl")
-    readRegionCSV(spark, basePath + "/region.tbl").write.parquet(baseOutput + "/region.tbl")
-    readSupplierCSV(spark, basePath + "/supplier.tbl").write.parquet(baseOutput + "/supplier.tbl")
+  private def setupWrite(
+      df: DataFrame, 
+      name: String,
+      coalesce: Map[String, Int],
+      repartition: Map[String, Int]): DataFrameWriter[Row] = {
+    val repart = (coalesce.get(name), repartition.get(name)) match {
```

Comment:
Every change I make to the benchmarks is done in triplicate at this point so I have been having the same thoughts. Initially, I thought it was best to keep them all independent but it would make my life much easier if we could start to move more code into the common module. We could even go further and have one benchmark runner and one main method and have it delegate to traits that the individual benchmarks implement.

Reply:
I filed https://github.com/NVIDIA/spark-rapids/issues/945 to discuss the larger refactor, but I'll move the common code in this PR into BenchUtils.